app_server/app/util/Utility.js

```diff
@@ -137,3 +137,27 @@ exports.changeToUnixTime = function(obj) {
 
   return obj;
 };
+
+var addValidationDetailsForJoi =
+/**
+ * Adds details for validation errors through validate's failAction property
+ * Assumes validation is done through joi
+ */
+exports.addValidationDetailsForJoi = function(request, reply, source, error) {
+  var errors = error.data.details.map(function(err) {
+    for (var key in err.context) {
+      if (err.context.hasOwnProperty(key) &&
+          err.context[key] && err.context[key].toString) {
```

Comment:
is there a case where `err.context.hasOwnProperty(key)` is true while `err.context[key]` will return false